tests/common/utils/platform/iot_test_platform_threads.c

```diff
@@ -0,0 +1,387 @@
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief helper function for testing thread creation
+ */
+void threadTestFunction(void * param)
+{
+    *( uint32_t * ) param = 4321;
+}
+
+TEST( UTIL_Platform_Threads, IotThreads_CreateDetachedThread )
+{
+    static uint32_t attrData = 1234;
+    Iot_CreateDetachedThread( threadTestFunction, &attrData, 5, 3072 );
+  
+    /* wait for the thread to set this */
+    for ( int i = 0 ; i<10 ; i++ )
+    {
+        /* If we see the value has changed we stop waiting */
+        if ( attrData != 1234 )
+           break;
+
+        /* Else we wait for 100ms here */
+        vTaskDelay( configTICK_RATE_HZ / 10 );
+    }
+    
+    TEST_ASSERT_EQUAL( 4321,  attrData );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief helper function for testing thread priority
+ */
+void threadPriorityTestFunction(void * param)
+{
+    TaskStatus_t xTaskDetails;
+
+    /* Use the handle to obtain further information about the task. */
+    vTaskGetInfo( /* The handle of the task being queried. */
+                  NULL,
+                  /* The TaskStatus_t structure to complete with information
+                  on xTask. */
+                  &xTaskDetails,
+                  /* Include the stack high water mark value in the
+                  TaskStatus_t structure. */
+                  pdTRUE,
+                  /* Include the task state in the TaskStatus_t structure. */
+                  pdTRUE );
+
+    *( int32_t * ) param = xTaskDetails.uxCurrentPriority;
+}
+
+TEST( UTIL_Platform_Threads, IotThreads_ThreadPriority )
+{
+    static int32_t attrData = -1;
+
+    /* Create thread with priority 0 */
+    Iot_CreateDetachedThread( threadPriorityTestFunction, &attrData, 0, 3072 );
+  
+    /* Wait for the thread to complete before checking result*/
+    while( attrData == -1 ) 
+    {
+        vTaskDelay( 1 );
+    }
+    
+    TEST_ASSERT_EQUAL( 0,  attrData );
+    printf ("Expected Pri = 0, actual = %d\r\n", attrData );
+    attrData = -1;
+
+    /* Create thread with priority 5 */
+    Iot_CreateDetachedThread( threadPriorityTestFunction, &attrData, 5, 3072 );
+  
+    /* Wait for the thread to complete before checking result*/
+    while( attrData == -1 ) 
+    {
+        vTaskDelay( 1 );
+    }
+    
+    printf ("Expected Pri = 5, actual = %d\r\n", attrData );
+    TEST_ASSERT_EQUAL( 5,  attrData );
+    attrData = -1;
+
+    /* Create thread with priority 5 */
+    Iot_CreateDetachedThread( threadPriorityTestFunction, &attrData, 7, 3072 );
+  
+    /* Wait for the thread to complete before checking result*/
+    while( attrData == -1 ) 
+    {
+        vTaskDelay( 1 );
+    }
+    
+    printf ("Expected Pri = 7, actual = %d\r\n", attrData );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief helper function for testing thread priority
+ */
+void threadStackSizeTestFunction(void * param)
```

Comment:
This test is very "iffy" as there is no easy way to get the stack size from FreeRTOS. I improvised a little but I think this is very hacky. Is there value in this test or do we just loose it? I added this test when the ESP code failed because the stack size I asked for was not what I received...

See the code in the threads_afr.c file handling defaults. When we change that to be a config parameter it is possible to create failing scenarios through configuration, which will be caught by this test.